hack/update-clientset.sh

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# https://github.com/kubernetes/code-generator provides generator code to generate a custom typed
+# and versioned client for custom API types similar to what https://github.com/kubernetes/client-go
+# provides for core types. This script generates such a client for the Gateway API types, in service of any
+# projects that need them.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+if [[ "${1:-stable}" == "experimental" ]]; then
+  readonly API_PATH="apisx"
+else
+  readonly API_PATH="apis"
+fi
+
+readonly SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE}")"/.. && pwd)"
+
+if [[ "${VERIFY_CODEGEN:-}" == "true" ]]; then
+  echo "Running in verification mode"
+  readonly VERIFY_FLAG="--verify-only"
+fi
+
+readonly COMMON_FLAGS="${VERIFY_FLAG:-} --go-header-file ${SCRIPT_ROOT}/hack/boilerplate/boilerplate.generatego.txt"
+
+readonly APIS_PKG=sigs.k8s.io/gateway-api
+readonly CLIENTSET_NAME=versioned
+readonly CLIENTSET_PKG_NAME=clientset
+readonly VERSIONS=($(find ./${API_PATH} -maxdepth 1 -name "v*" -exec bash -c 'basename {}' \; | xargs))
+
+if [[ "${1:-stable}" == "experimental" ]]; then
+    readonly OUTPUT_DIR=pkg/clientx
+    readonly OUTPUT_PKG=sigs.k8s.io/gateway-api/pkg/clientx
+else
+    readonly OUTPUT_DIR=pkg/client
+    readonly OUTPUT_PKG=sigs.k8s.io/gateway-api/pkg/client
+fi
```

Comment:
I'm a little curious if a single clientset would be better here. With that said, this will likely become clearer as we start working with ListenerSet. I'm fine with this as a starting point.

Reply:
yeah this can be adjusted later